### PR TITLE
Only publish new version when changes are made

### PR DESCRIPTION
**Pull request by niklaskors**
We could improve the git diff.. so that it doesn't throw an error but it'll work for now.
Already published this to relivecc/packages#gitpkg

### CODE REVIEW

File: src/tasks/Publish/upload-package.js [hunk 1]
```diff
@@ -9,18 +9,55 @@ export default async function uploadPackage(pkg, pkgPath, registry) {
...
+      console.log("No changes detected");
+      return;
+    }
+
+    try {
+      await execLikeShell(
+        `git push --delete origin ${gitpkgPackageName}`,
+        pkgTempDirPkg
+      );
+      await execLikeShell(`git tag -d ${gitpkgPackageName}`, pkgTempDirPkg);
```
Comment (YousefED) [inline src/tasks/Publish/upload-package.js:59]: Why is this here? And why is `await execLikeShell(`git tag ${gitpkgPackageName}`, pkgTempDirPkg);` moved down?

Shouldn't the only change necessary be the `git diff ${gitpkgPackageName} --quiet` you added?

Comment (niklaskors) [reply src/tasks/Publish/upload-package.js:59]: So in order to diff the changes we need the tag inside our local git. And for that I needed to change the order of commands a bit:
1. Fetch all tags from remote
2. Check if the tag already exists (if not, skip step 3&4)
3. If the tag exists we can diff the new local changes with the tag
4. Because we fetched all tags from the remote we need to delete the remote and local tag otherwise we cannot publish on the same tag.
5. Push

Comment (YousefED) [reply src/tasks/Publish/upload-package.js:59]: got it, maybe copy your step 1-4 explanation to code. but nice job!
